### PR TITLE
Jar support updated

### PR DESCRIPTION
**Pull request by htorrence**
Working off of Ken's PR (#8 ),  this expands the logic to work for jars as well to extend usefulness to our Scala friends. In the process I also added logging and updated dependencies. Thanks @kdallmeyer-sr for getting this started!

Pull Request Checklist
 - [x] Pull request is based on WIP branch
 - [x] Pull request includes a description of the change and the reason behind it
 - [x] `pytest` passes locally
 - [x] `flake8 apparate tests` passes locally
 - [x] CHANGELOG has been updated
 - [x] Version in `_version.py` has been updated
 - [x] README has been updated (if applicable)

### CODE REVIEW

File: apparate/cli_commands.py [hunk 1]
```diff
@@ -66,7 +74,14 @@ def upload(path, token, folder):
     token = _resolve_input(token, 'token', 'token', config)
     folder = _resolve_input(folder, 'folder', 'prod_folder', config)
 
-    update_databricks(path, token, folder, update_jobs=False, cleanup=False)
+    update_databricks(
+        logger,
```
Comment (jscottcronin) [inline apparate/cli_commands.py]: just curious why we have to pass the logger in. Does it not work by importing the logger in update_databricks file and executing log statements?

File: apparate/update_databricks_library.py [hunk 2]
```diff
@@ -37,18 +39,119 @@ def __str__(self):
...
+            self.filename = filename
+            self.library_name = match.group(1)
+            self.version = match.group(2)
+            self.major_version = match.group(3)
+            self.minor_version = match.group(4)
+            self.suffix = match.group(5)
+            if self.suffix == 'jar':
+                self.lib_type = 'java-jar'
+            elif self.suffix == 'egg':
+                self.lib_type = 'python-egg'
```
Comment (jscottcronin) [inline apparate/update_databricks_library.py]: this is much more readable than the prior logic. 💯

File: apparate/update_databricks_library.py [hunk 3]
```diff
@@ -107,41 +210,42 @@ def get_job_list(library_name, major_version, library_mapping, token, host):
...
         for job in res.json()['jobs']:
+            logger.debug('job: {}'.format(job['settings']['name']))
             if 'libraries' in job['settings'].keys():
                 for library in job['settings']['libraries']:
-                    if 'egg' in library.keys():
+                    if match.suffix in library.keys():
                         try:  # if in prod_folder, mapping turns uri into name
-                            job_library_uri = library['egg'].split('/')[-1]
-                            job_library = library_mapping[job_library_uri]
+                            job_library_uri = basename(library[match.suffix])
```
Comment (jscottcronin) [inline apparate/update_databricks_library.py]: [not necessary] I sometimes find it beneficial to use `os.path.basename` as opposed to importing `basename` which reminds us that we are using stdlib functions, but that is purely personal preference

File: apparate/cli_commands.py (hunk 1 above)
Comment (htorrence) [reply apparate/cli_commands.py]: this is me not understanding how logging propagates configs - I think I understand what's happening better now. Does this look better? Are there things you would do differently with the logging here?

File: apparate/update_databricks_library.py (hunk 2 above)
Comment (htorrence) [reply apparate/update_databricks_library.py]: Agreed! Thanks to Ken for setting this up.

File: apparate/update_databricks_library.py (hunk 3 above)
Comment (htorrence) [reply apparate/update_databricks_library.py]: That makes sense, but I think I'm going to stick with the less verbose option in this particular case.

File: apparate/cli_commands.py (hunk 1 above)
Comment (htorrence) [reply apparate/cli_commands.py]: Hmm, actually I wasn't able to get the click verbosity setting to propagate when importing the logger. How would you imagine doing it?

**Comment by htorrence** [comment]
So sadly the library upload API endpoint does not consider the file extension, so it's impossible upload an egg and a jar with the same name/version to the same folder via the API. Instead I just clarified the error message to account for this case.

**Comment by htorrence** [comment]
@jscottcronin I think this is ready for another look

File: apparate/cli_commands.py (hunk 1 above)
Comment (jscottcronin) [reply apparate/cli_commands.py]: I looked into a little bit, and did not realize the logger was coming from click_log. I could not think of a simple solution, so I suggest we just skip this for now. Perhaps a contributor can add a fix :)

**Pull request merged**